### PR TITLE
Enhancement of Test Coverage and Code Improvements in Container Scan Package

### PR DESCRIPTION
**Pull request by cbrom**
## PR Type:
Enhancement

___
## PR Description:
This PR primarily focuses on enhancing the test coverage and improving the code quality of the 'containerscan' package. The main changes include:
- Addition of new test cases for the 'CalculateFixed' function.
- Removal of the 'GetFilesByPackage' function and its associated tests, as it was not used anywhere except in tests.
- Addition of a doc comment for the exported function 'GetPackageNames' and a basic test for it.
- Addition of doc strings for exported functions in 'datastructuresmethods.go'.
- Addition of tests for 'ScanResultReport.Validate' function.
- Removal of 'ToFlatVulnerabilities' function as there is no reference anywhere.
- Addition of 'ElasticContainerScanSummaryResult Validate' function doc string and tests for it.
- Coverage test for 'ScanResultReport AsFNVHash' function.
- Update in 'go.sum' file.

___
## PR Main Files Walkthrough:
<details> <summary>files:</summary>

- `core/pkg/containerscan/containerscan_test.go`: Added new test cases for 'CalculateFixed', 'ScanResultReport.Validate', 'ScanElasticContainerScanSummaryResultValidate' functions. Updated the test for 'GetPackagesNames' function.
- `core/pkg/containerscan/datastructures.go`: Added a detailed docstring for the 'CalculateFixed' function.
- `core/pkg/containerscan/datastructuresmethods.go`: Removed the 'GetFilesByPackage' function. Added a docstring for 'GetPackagesNames' function. Added docstrings for 'GetDesignatorsNContext', 'Validate', and 'IsRCE' functions.
- `core/pkg/containerscan/elasticadapters.go`: Removed the 'ToFlatVulnerabilities' function. The 'Summarize' function remains unchanged.
- `core/pkg/containerscan/elasticdatastructures.go`: Added a 'Validate' function to check if the 'ElasticContainerScanSummaryResult' object is valid.
- `go.sum`: Updated the dependencies.
</details>

### CODE REVIEW

**Comment by cbrom** [comment]
@matthyx This one is ready for review.

File: core/pkg/containerscan/elasticadapters.go [hunk 1]
```diff
@@ -5,55 +5,7 @@ import (
 	cautils "github.com/armosec/utils-k8s-go/armometadata"
 )
 
-// ToFlatVulnerabilities - returnsgit p
-func (scanresult *ScanResultReport) ToFlatVulnerabilities() []*ElasticContainerVulnerabilityResult {
```
Comment (matthyx) [inline core/pkg/containerscan/elasticadapters.go:9]: I am surprised by this one... did you cover that case elsewhere and I didn't notice?

Comment (cbrom) [reply core/pkg/containerscan/elasticadapters.go:9]: I removed it because I couldn't find any function call and I thought it was obsolete.

**Review by matthyx — APPROVED** [review]

**Pull request merged**